ssh-pairing-server.c

```diff
@@ -85,6 +151,34 @@ int main(int argc, char *argv[])
 			}
 		} else if (msg_type == SSH_REQUEST_AUTH && msg_subtype == SSH_AUTH_METHOD_INTERACTIVE) {
 			char *msg = NULL;
+			/*
+			 * I nearly overlooked that this message goes out to
+			 * the connecting client, not to stdout.
+			 *
+			 * Providing any information to a yet unauthenticated
+			 * client can quickly become problematic. It looks
+			 * like we are only reporting back what the client
+			 * already knows: the amount of public keys it sent to
+			 * us, the username it used and the IP address it
+			 * uses.
+			 *
+			 * At least the latter (the IP address) could be a
+			 * small information leak in some setups where some
+			 * form of proxy / forwarding / NAT whatever is used.
+			 * In this case the IP address would leak some network
+			 * topology detail to unauthenticated clients.
+			 *
+			 * I'm not sure if there is much value in sending this
+			 * out to the unauthenticated client. The user that
+			 * rightfully uses this tool has two interaction
+			 * points: The server side and the client side. It
+			 * could be more clean and more secure to restrict
+			 * any kind of informational output messages to the
+			 * server side.
+			 *
+			 * Then you could also drop this INTERACTIVE auth path
```

Comment:
Some kind of response that indicates a successful reception is IMO important, as it tells the client that it was the "chosen one" for the enrollment. I can remove the `from %s@%s` part though if you think that it's not useful or even dangerous.

Reply:
So you see this as part of the verification process. It could even work out ... the client verifies the server's host key, so at this point the messages returned from the server should be trusted. No man-in-the-middle or IP-spoofer should be able to get to this point.

In that case this is a vital part of the verification process and the user needs to be clearly instructed to look out for this message. It might even make sense to report back the fingerprints of the individual keys seen by the server.

The `from %s@%s` part can be safely dropped I guess it doesn't add much value for the legitimate client but could pose a small information leak for non-legit clients.

Reply:
Ah, so repurposing this channel for public key exchange is a bit hairy after all. A condensed message will have to suffice then I guess.